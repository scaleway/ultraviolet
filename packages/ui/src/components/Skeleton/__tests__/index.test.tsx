import { renderWithTheme } from '@utils/test'
import { describe, expect, test } from 'vitest'
import { Skeleton, skeletonTypes } from '..'

describe('skeleton', () => {
  test('renders default variant', () => {
    const { asFragment } = renderWithTheme(<Skeleton />)
    expect(asFragment()).toMatchSnapshot()
  })

  test.each(skeletonTypes)('renders correctly with type="%s"', type => {
    const { asFragment } = renderWithTheme(<Skeleton variant={type} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
