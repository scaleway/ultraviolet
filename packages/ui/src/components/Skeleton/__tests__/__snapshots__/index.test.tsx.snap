// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`skeleton > renders correctly with type="block" 1`] = `
<DocumentFragment>
  <div
    data-testid="testing"
  >
    <div
      aria-busy="true"
      aria-live="polite"
      class="styles__c2hb451"
    >
      <ul
        class="stylesVariants__fr4c0t4"
      >
        <li
          class="stylesVariants__fr4c0t3"
        >
          <div
            class="stylesVariants__fr4c0tc"
          />
          <div
            class="stylesVariants__fr4c0td"
            style="--fr4c0t1: 7.5rem;"
          />
        </li>
        <li
          class="stylesVariants__fr4c0t3"
        >
          <div
            class="stylesVariants__fr4c0tc"
          />
          <div
            class="stylesVariants__fr4c0td"
            style="--fr4c0t1: 7.5rem;"
          />
        </li>
        <li
          class="stylesVariants__fr4c0t3"
        >
          <div
            class="stylesVariants__fr4c0tc"
          />
          <div
            class="stylesVariants__fr4c0td"
            style="--fr4c0t1: 7.5rem;"
          />
        </li>
      </ul>
      <div
        class="styles__c2hb452"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`skeleton > renders correctly with type="blocks" 1`] = `
<DocumentFragment>
  <div
    data-testid="testing"
  >
    <div
      aria-busy="true"
      aria-live="polite"
      class="styles__c2hb451"
    >
      <div
        class="stylesVariants__fr4c0t5"
        style="--fr4c0t0: 4;"
      >
        <div
          class="stylesVariants__fr4c0t6"
        >
          <div
            class="stylesVariants__fr4c0tc"
          />
          <div
            class="stylesVariants__fr4c0td"
            style="--fr4c0t1: 7.5rem;"
          />
        </div>
        <div
          class="stylesVariants__fr4c0t6"
        >
          <div
            class="stylesVariants__fr4c0tc"
          />
          <div
            class="stylesVariants__fr4c0td"
            style="--fr4c0t1: 7.5rem;"
          />
        </div>
        <div
          class="stylesVariants__fr4c0t6"
        >
          <div
            class="stylesVariants__fr4c0tc"
          />
          <div
            class="stylesVariants__fr4c0td"
            style="--fr4c0t1: 7.5rem;"
          />
        </div>
        <div
          class="stylesVariants__fr4c0t6"
        >
          <div
            class="stylesVariants__fr4c0tc"
          />
          <div
            class="stylesVariants__fr4c0td"
            style="--fr4c0t1: 7.5rem;"
          />
        </div>
        <div
          class="stylesVariants__fr4c0t6"
        >
          <div
            class="stylesVariants__fr4c0tc"
          />
          <div
            class="stylesVariants__fr4c0td"
            style="--fr4c0t1: 7.5rem;"
          />
        </div>
        <div
          class="stylesVariants__fr4c0t6"
        >
          <div
            class="stylesVariants__fr4c0tc"
          />
          <div
            class="stylesVariants__fr4c0td"
            style="--fr4c0t1: 7.5rem;"
          />
        </div>
        <div
          class="stylesVariants__fr4c0t6"
        >
          <div
            class="stylesVariants__fr4c0tc"
          />
          <div
            class="stylesVariants__fr4c0td"
            style="--fr4c0t1: 7.5rem;"
          />
        </div>
        <div
          class="stylesVariants__fr4c0t6"
        >
          <div
            class="stylesVariants__fr4c0tc"
          />
          <div
            class="stylesVariants__fr4c0td"
            style="--fr4c0t1: 7.5rem;"
          />
        </div>
      </div>
      <div
        class="styles__c2hb452"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`skeleton > renders correctly with type="box" 1`] = `
<DocumentFragment>
  <div
    data-testid="testing"
  >
    <div
      aria-busy="true"
      aria-live="polite"
      class="styles__c2hb451"
    >
      <div
        class="stylesVariants__fr4c0t5"
        style="--fr4c0t0: 4;"
      >
        <div
          class="stylesVariants__fr4c0t7"
        >
          <div
            class="stylesVariants__fr4c0tc"
          />
          <div
            class="stylesVariants__fr4c0td"
            style="--fr4c0t1: 7.5rem;"
          />
        </div>
        <div
          class="stylesVariants__fr4c0t7"
        >
          <div
            class="stylesVariants__fr4c0tc"
          />
          <div
            class="stylesVariants__fr4c0td"
            style="--fr4c0t1: 7.5rem;"
          />
        </div>
        <div
          class="stylesVariants__fr4c0t7"
        >
          <div
            class="stylesVariants__fr4c0tc"
          />
          <div
            class="stylesVariants__fr4c0td"
            style="--fr4c0t1: 7.5rem;"
          />
        </div>
        <div
          class="stylesVariants__fr4c0t7"
        >
          <div
            class="stylesVariants__fr4c0tc"
          />
          <div
            class="stylesVariants__fr4c0td"
            style="--fr4c0t1: 7.5rem;"
          />
        </div>
        <div
          class="stylesVariants__fr4c0t7"
        >
          <div
            class="stylesVariants__fr4c0tc"
          />
          <div
            class="stylesVariants__fr4c0td"
            style="--fr4c0t1: 7.5rem;"
          />
        </div>
        <div
          class="stylesVariants__fr4c0t7"
        >
          <div
            class="stylesVariants__fr4c0tc"
          />
          <div
            class="stylesVariants__fr4c0td"
            style="--fr4c0t1: 7.5rem;"
          />
        </div>
        <div
          class="stylesVariants__fr4c0t7"
        >
          <div
            class="stylesVariants__fr4c0tc"
          />
          <div
            class="stylesVariants__fr4c0td"
            style="--fr4c0t1: 7.5rem;"
          />
        </div>
        <div
          class="stylesVariants__fr4c0t7"
        >
          <div
            class="stylesVariants__fr4c0tc"
          />
          <div
            class="stylesVariants__fr4c0td"
            style="--fr4c0t1: 7.5rem;"
          />
        </div>
      </div>
      <div
        class="styles__c2hb452"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`skeleton > renders correctly with type="donut" 1`] = `
<DocumentFragment>
  <div
    data-testid="testing"
  >
    <div
      aria-busy="true"
      aria-live="polite"
      class="styles__c2hb451"
    >
      <div
        class="stylesVariants__fr4c0t8"
      >
        <svg
          class="stylesVariants__fr4c0t9"
        >
          <circle
            class="stylesVariants__fr4c0ta"
            cx="6.4375rem"
            cy="6.4375rem"
            r="90"
          />
        </svg>
        <ul
          class="stylesVariants__fr4c0tb"
        >
          <li>
            <div
              class="stylesVariants__fr4c0td"
              style="--fr4c0t1: 7.5rem;"
            />
          </li>
          <li>
            <div
              class="stylesVariants__fr4c0td"
              style="--fr4c0t1: 7.5rem;"
            />
          </li>
          <li>
            <div
              class="stylesVariants__fr4c0td"
              style="--fr4c0t1: 7.5rem;"
            />
          </li>
          <li>
            <div
              class="stylesVariants__fr4c0td"
              style="--fr4c0t1: 7.5rem;"
            />
          </li>
        </ul>
      </div>
      <div
        class="styles__c2hb452"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`skeleton > renders correctly with type="line" 1`] = `
<DocumentFragment>
  <div
    data-testid="testing"
  >
    <div
      aria-busy="true"
      aria-live="polite"
      class="styles__c2hb451"
    >
      <div
        class="stylesVariants__fr4c0td"
        style="--fr4c0t1: 7.5rem;"
      />
      <div
        class="styles__c2hb452"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`skeleton > renders correctly with type="list" 1`] = `
<DocumentFragment>
  <div
    data-testid="testing"
  >
    <div
      aria-busy="true"
      aria-live="polite"
      class="styles__c2hb451"
    >
      <ul
        class="stylesVariants__fr4c0tg"
      >
        <li
          class="stylesVariants__fr4c0te"
        >
          <div
            class="stylesVariants__fr4c0tf"
          >
            <div
              class="stylesVariants__fr4c0td"
              style="--fr4c0t1: 7.5rem;"
            />
          </div>
          <div
            class="stylesVariants__fr4c0tf"
          >
            <div
              class="stylesVariants__fr4c0td"
              style="--fr4c0t1: 7.5rem;"
            />
          </div>
          <div
            class="stylesVariants__fr4c0tf"
          >
            <div
              class="stylesVariants__fr4c0td"
              style="--fr4c0t1: 7.5rem;"
            />
          </div>
        </li>
        <li
          class="stylesVariants__fr4c0te"
        >
          <div
            class="stylesVariants__fr4c0tf"
          >
            <div
              class="stylesVariants__fr4c0td"
              style="--fr4c0t1: 7.5rem;"
            />
          </div>
          <div
            class="stylesVariants__fr4c0tf"
          >
            <div
              class="stylesVariants__fr4c0td"
              style="--fr4c0t1: 7.5rem;"
            />
          </div>
          <div
            class="stylesVariants__fr4c0tf"
          >
            <div
              class="stylesVariants__fr4c0td"
              style="--fr4c0t1: 7.5rem;"
            />
          </div>
        </li>
        <li
          class="stylesVariants__fr4c0te"
        >
          <div
            class="stylesVariants__fr4c0tf"
          >
            <div
              class="stylesVariants__fr4c0td"
              style="--fr4c0t1: 7.5rem;"
            />
          </div>
          <div
            class="stylesVariants__fr4c0tf"
          >
            <div
              class="stylesVariants__fr4c0td"
              style="--fr4c0t1: 7.5rem;"
            />
          </div>
          <div
            class="stylesVariants__fr4c0tf"
          >
            <div
              class="stylesVariants__fr4c0td"
              style="--fr4c0t1: 7.5rem;"
            />
          </div>
        </li>
      </ul>
      <div
        class="styles__c2hb452"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`skeleton > renders correctly with type="slider" 1`] = `
<DocumentFragment>
  <div
    data-testid="testing"
  >
    <div
      aria-busy="true"
      aria-live="polite"
      class="styles__c2hb451"
    >
      <div
        class="stylesVariants__fr4c0th"
        style="--fr4c0t2: 4;"
      >
        <div
          class="stylesVariants__fr4c0ti"
        >
          <div
            class="stylesVariants__fr4c0tj"
          />
        </div>
        <div
          class="stylesVariants__fr4c0ti"
        >
          <div
            class="stylesVariants__fr4c0tj"
          />
        </div>
        <div
          class="stylesVariants__fr4c0ti"
        >
          <div
            class="stylesVariants__fr4c0tj"
          />
        </div>
        <div
          class="stylesVariants__fr4c0ti"
        >
          <div
            class="stylesVariants__fr4c0tj"
          />
        </div>
      </div>
      <div
        class="styles__c2hb452"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`skeleton > renders correctly with type="square" 1`] = `
<DocumentFragment>
  <div
    data-testid="testing"
  >
    <div
      aria-busy="true"
      aria-live="polite"
      class="styles__c2hb451"
    >
      <div
        class="stylesVariants__fr4c0tk"
      />
      <div
        class="styles__c2hb452"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`skeleton > renders default variant 1`] = `
<DocumentFragment>
  <div
    data-testid="testing"
  >
    <div
      aria-busy="true"
      aria-live="polite"
      class="styles__c2hb451"
    >
      <div
        class="stylesVariants__fr4c0t5"
        style="--fr4c0t0: 4;"
      >
        <div
          class="stylesVariants__fr4c0t6"
        >
          <div
            class="stylesVariants__fr4c0tc"
          />
          <div
            class="stylesVariants__fr4c0td"
            style="--fr4c0t1: 7.5rem;"
          />
        </div>
        <div
          class="stylesVariants__fr4c0t6"
        >
          <div
            class="stylesVariants__fr4c0tc"
          />
          <div
            class="stylesVariants__fr4c0td"
            style="--fr4c0t1: 7.5rem;"
          />
        </div>
        <div
          class="stylesVariants__fr4c0t6"
        >
          <div
            class="stylesVariants__fr4c0tc"
          />
          <div
            class="stylesVariants__fr4c0td"
            style="--fr4c0t1: 7.5rem;"
          />
        </div>
        <div
          class="stylesVariants__fr4c0t6"
        >
          <div
            class="stylesVariants__fr4c0tc"
          />
          <div
            class="stylesVariants__fr4c0td"
            style="--fr4c0t1: 7.5rem;"
          />
        </div>
        <div
          class="stylesVariants__fr4c0t6"
        >
          <div
            class="stylesVariants__fr4c0tc"
          />
          <div
            class="stylesVariants__fr4c0td"
            style="--fr4c0t1: 7.5rem;"
          />
        </div>
        <div
          class="stylesVariants__fr4c0t6"
        >
          <div
            class="stylesVariants__fr4c0tc"
          />
          <div
            class="stylesVariants__fr4c0td"
            style="--fr4c0t1: 7.5rem;"
          />
        </div>
        <div
          class="stylesVariants__fr4c0t6"
        >
          <div
            class="stylesVariants__fr4c0tc"
          />
          <div
            class="stylesVariants__fr4c0td"
            style="--fr4c0t1: 7.5rem;"
          />
        </div>
        <div
          class="stylesVariants__fr4c0t6"
        >
          <div
            class="stylesVariants__fr4c0tc"
          />
          <div
            class="stylesVariants__fr4c0td"
            style="--fr4c0t1: 7.5rem;"
          />
        </div>
      </div>
      <div
        class="styles__c2hb452"
      />
    </div>
  </div>
</DocumentFragment>
`;
